pkg/deploy/deploy.go

```diff
@@ -288,9 +288,19 @@ func manageResource(ctx context.Context, cli client.Client, obj *unstructured.Un
 		return nil
 	}
 
-	// do not reconcile kserve resource with annotation "opendatahub.io/managed: false"
-	if found.GetAnnotations()["opendatahub.io/managed"] == "false" && componentName == "kserve" {
-		return nil
+	// Kserve specific workflow:
+	// TODO: Remove this when we have generalize custom config requirements across all components
+	if componentName == "kserve" {
+		// do not reconcile kserve resource with annotation "opendatahub.io/managed: false"
+		if found.GetAnnotations()["opendatahub.io/managed"] == "false" {
+			return nil
+		}
+		// do not patch resources field in Kserve deployment i.e allows users to update resources field
+		if found.GetKind() == "Deployment" {
+			if _, err := removeResourcesFromDeployment(obj); err != nil {
+				return err
```

Comment:
Do you really want to fail if cannot patch it?

Reply:
Thats a good question, we probably should not fail here. But then it will continue with the patch and users will loose their configuration

Reply:
> Thats a good question, we probably should not fail here. But then it will continue with the patch and users will loose their configuration

Should you return nil then if you want to avoid updating just that resource (with a loud message probably)?